src/NumSharp.Core/Logic/np.logical_and.cs

```diff
@@ -16,7 +16,70 @@ public static partial class np
         /// <remarks>https://docs.scipy.org/doc/numpy/reference/generated/numpy.logical_and.html</remarks>
         public static unsafe NDArray<bool> logical_and(NDArray lhs, NDArray rhs)
         {
-            if (lhs.typecode == NPTypeCode.Boolean && rhs.typecode == NPTypeCode.Boolean)
+#if _REGEN
+            if(lhs.typecode != rhs.typecode)
+            {
+                throw new NotImplementedException("please make sure operands have the same data type");
```

Comment:
They do not have to be of the same dtype.
Like all operators, this has to support all dtypes combinations.